pkg/controller/container-runtime-config/container_runtime_config_bootstrap.go

```diff
@@ -40,7 +40,7 @@ func RunContainerRuntimeBootstrap(templateDir string, crconfigs []*mcfgv1.Contai
 
 			var configFileList []generatedConfigFile
 			ctrcfg := cfg.Spec.ContainerRuntimeConfig
-			if !ctrcfg.OverlaySize.IsZero() {
+			if ctrcfg.OverlaySize != nil {
```

Comment:
Applies to all changes:

Doesn’t this change the semantics of previously-created CRs? If I understand the situation correctly, it’s possible that users create the CRs with these fields missing, and something (`patchContainerRuntimeConfigs`??) updates them and adds `"0"` field values.

In the old version, those 0 values were treated as missing, i.e. the system did exactly what the users wanted, just in a confusing way; with this PR, wouldn’t the `0` values actually start being applied?

I suppose one way to tell would be to add (uint? e2e?) tests with CRs of that kind, and ensure that both empty values and `"0"` values are treated as missing. (Or maybe those tests already exist? I didn’t find them though a quick grep but I didn’t spend much time looking.)

Reply:
0 values will not be applied in this PR. The drop-in files will not be created if `0` values. https://github.com/openshift/machine-config-operator/pull/3815/files#diff-b2ef9db0d76c6145ec1bb5310134037bae64572ab47464d3223745c2a410d3e3R301

Reply:
Oops, my mistake, I didn’t notice that logic.

`OverlaySize` will, AFAICS, still cause `mergeConfigChanges` to be called, though with no edits. I guess that doesn’t make a difference.

---

Aesthetically I’d prefer for the code to consistently use a single condition, so that we just have “set to a relevant value / not set to a relevant value”, not “unset / set to zero / set to non-zero” to track.

But that’s weak a code style preference, to be decided by MCO maintainers.